packages/arch/packages/tooltip/src/index.js

```diff
@@ -57,118 +56,122 @@ const showTooltip = (fn, defaultDelay) => {
-  };
-
-  handleMouseEnter = () => {
-    this.cancelPendingSetState();
-
-    if (this.state.isVisible) {
-      return;
-    }
-    if (this.props.hideOnMouseDown && this.ref.current) {
-      this.ref.current.addEventListener('mousedown', this.cancel, LISTENER_OPTIONS);
-    }
-    if (this.props.hideOnKeyDown) {
-      document.addEventListener('keydown', this.cancel, LISTENER_OPTIONS);
-    }
+    let cancelPendingSetState = () => {};
+
+    const cancel = () => {
+      cancelPendingSetState();
+      setIsVisible(false);
+    };
+
+    const handleMouseEnter = () => {
+      cancelPendingSetState();
+
+      if (isVisible) {
+        return;
+      }
+
+      if (hideOnMouseDown && target) {
+        target.addEventListener('mousedown', cancel, LISTENER_OPTIONS);
+      }
+
+      if (hideOnKeyDown) {
+        document.addEventListener('keydown', cancel, LISTENER_OPTIONS);
+      }
+
+      cancelPendingSetState = showTooltip(() => setIsVisible(true), delay);
+    };
+
+    const handleMouseLeave = () => {
+      cancelPendingSetState();
+
+      if (!isVisible) {
+        return;
+      }
+
+      if (hideOnMouseDown && target) {
+        target.removeEventListener('mousedown', cancel, LISTENER_OPTIONS);
+      }
+
+      if (hideOnKeyDown) {
+        document.removeEventListener('keydown', cancel, LISTENER_OPTIONS);
+      }
+
+      cancelPendingSetState = hideTooltip(() => setIsVisible(false), delay);
+    };
+
+    target.addEventListener('mouseenter', handleMouseEnter, LISTENER_OPTIONS);
+    target.addEventListener('mouseleave', handleMouseLeave, LISTENER_OPTIONS);
+
+    return () => {
+      cancelPendingSetState();
+
+      if (target) {
+        target.removeEventListener('mouseenter', handleMouseEnter, LISTENER_OPTIONS);
+        target.removeEventListener('mouseleave', handleMouseLeave, LISTENER_OPTIONS);
+      }
+    };
+  }, [isVisible]);
+
+  return (
+    <Fragment>
+      {children(ref)}
+
+      <TransitionProvider isOpen={isVisible} onEntered={onShow} onExited={onHide}>
+        {transitionState => (
+          <TooltipPositioner
+            targetNode={ref.current}
+            placement={placement}
+            className={className}
+            style={fade(transitionState)}
+          >
+            {content}
+          </TooltipPositioner>
+        )}
+      </TransitionProvider>
+    </Fragment>
+  );
+};
 
-    this.cancelPendingSetState = showTooltip(immediatelyShow => {
-      this.setState({
-        isVisible: true,
-        immediatelyShow,
-      });
-    }, this.props.delay);
-  };
-  handleMouseLeave = () => {
-    this.cancelPendingSetState();
-
-    if (!this.state.isVisible) {
-      return;
-    }
-    if (this.props.hideOnMouseDown && this.ref.current) {
-      this.ref.current.removeEventListener('mousedown', this.cancel, LISTENER_OPTIONS);
-    }
-    if (this.props.hideOnKeyDown) {
-      document.removeEventListener('keydown', this.cancel, LISTENER_OPTIONS);
-    }
+Tooltip.defaultProps = {
```

Comment:
The React team is deprecating `defaultProps`:

https://github.com/reactjs/rfcs/blob/createlement-rfc/text/0000-create-element-changes.md

It may be worthwhile pulling these in as default argument values.

Reply:
We've got this in a bunch of places... I agree but I think we could defer this to another PR for now.

Reply:
Yes, I'll handle that separately at some point.